code/ensemble/utils/use-ensemble-criteria.R

```diff
@@ -0,0 +1,60 @@
+# Load models according to inclusion criteria before ensembling
+# 
+# Steps:
+# Currently, models included based on having: 
+# 1. All quantiles
+# 2. 4 horizons 
+# 3. Not manually specified for exclusion 
+# 4. Not the hub ensemble
+# 
+# Returns a list:
+# - "forecasts" = tibble, all forecasts passing inclusion criteria
+# - "criteria" = tibble, model names and criteria assessment
+# 
+library(dplyr)
+library(tibble)
+
+use_ensemble_criteria <- function(forecasts = 
+                                    covidHubUtils::load_forecasts(source = "local_hub_repo",
+                                                                  hub_repo_path = here(),
+                                                                  hub = "ECDC",
+                                                                  forecast_dates = forecast_date),
+                                  exclude_models = NULL,
+                                  team_name = "EuroCOVIDhub",
+                                  forecast_date = floor_date(today(), "week", 1)) {
+  
+  # 1. Identify models with all quantiles
+  quantiles <- round(c(0.01, 0.025, seq(0.05, 0.95, by = 0.05), 0.975, 0.99), 3)
+  
+  all_quantiles <- forecasts %>%
+    group_by(model, target_variable, location, target_end_date) %>%
+    summarise(n_quantiles = n(), 
+              .groups = "drop") %>%
+    filter(n_quantiles != length(quantile)) %>%
+    pull(model) %>%
```

Comment:
This still includes any model that passes the check for any of the target variables / locations / target_end_date. I think you'll need to keep the full information here (whether all_quantiles present, i.e. don't do the pull but return the data frame information by target and location).

The edge case we've had is where one model, at one date, for one target is missing some quantiles for one horizon. This model should be excluded from the ensemble for that target/location combination but included for all the other ones.

I think this could be done here by doing (unchecked so might be wrong):
```{r}
all_quantiles <- forecasts %>%
    group_by(model, target_variable, location, target_end_date) %>%
    summarise(all_quantiles_present = setequal(quantile, quantiles)) %>%
    group_by(model, target_variable, location) %>%
    summarise(all_quantiles_present_for_all_horizons = all(all_quantiles_present)))
```